src/amplitude/client.py

```diff
@@ -167,3 +170,10 @@ def shutdown(self):
         """Shutdown the client instance, not accepting new events, flush all events in buffer"""
         self.configuration.opt_out = True
         self.__timeline.shutdown()
+
+    def _register_on_exit(self):
```

Comment:
is it worth it to add a try catch so this method does not throw new errors that get surfaced to customers? if it does fail, then they'll probably see the old error

Reply:
In general, I believe it is safe. I can still add a try-catch block for better safety.

Reply:
@kevinpagtakhan add a block for it